docs/guide/vite.md

```diff
@@ -0,0 +1,383 @@
+:::
+
+### WebSocket Configuration
+
+By default, the HMR WebSocket server runs on your ApostropheCMS server port, typically port 3000. For cases where you need a separate port (like running behind certain proxy configurations), you can configure a custom port:
+
+<AposCodeBlock>
+
+```javascript
+'@apostrophecms/asset': {
+  options: {
+    hmrPort: 3001  // Sets a custom WebSocket server port
+  }
+}
+```
+  <template v-slot:caption>
+    app.js
+  </template>
+</AposCodeBlock>
+
+### Source Maps in Production
+
+For debugging production builds, you can enable source maps to see the original source code in your browser's developer tools:
+
+<AposCodeBlock>
+
+```javascript
+'@apostrophecms/asset': {
+  options: {
+    productionSourceMaps: true  // Exposes source maps in production
+  }
+}
+```
+  <template v-slot:caption>
+    app.js
+  </template>
+</AposCodeBlock>
+
+::: tip
+Source maps slightly increase your build size but are invaluable for debugging production issues. Consider your specific needs when enabling this option.
+:::
+
+## Extending Vite Configuration
+
+ApostropheCMS's Vite integration provides two methods to customize your Vite configuration. You can either configure it through your module's code or via a dedicated configuration file in your project root.
+
+### Method 1: Module Configuration
+
+Use the `build.vite` property in your module to extend Vite's configuration:
+<AposCodeBlock>
+
+```javascript
+module.exports = {
+  build: {
+    vite: {
+      myViteConfig: {
+        // Vite configuration options
+        define: {
+          __API_URL__: JSON.stringify('https://api.example.com'),
+          __ANALYTICS_ENDPOINT__: JSON.stringify(process.env.ANALYTICS_URL)
+        },
+        css: {
+          // CSS configuration options
+          preprocessorOptions: {
+            scss: {
+              additionalData: `$injectedColor: orange;`
+            }
+          }
+        },
+        optimizeDeps: {
+          // Dependencies optimization options
+          include: ['lodash', 'moment']
+        },
+        server: {
+          // Development server options
+          https: true,
+          port: 3000
+        }
+      }
+    }
+  }
+};
+```
+  <template v-slot:caption>
+    modules/analytics/index.js
+  </template>
+</AposCodeBlock>
+
+### Method 2: Project Configuration File
+
+For project-wide Vite settings, create either `apos.vite.config.js` (ESM) or `apos.vite.config.mjs` (CJS) in your project root:
+
+<AposCodeBlock>
+
+```javascript
+import { defineConfig } from '@apostrophecms/vite/vite';
+
+export default defineConfig({
+  // All standard Vite configuration options are supported
+  plugins: [
+    // Add Vite plugins
+    svgLoader({
+      svgoConfig: {/* custom SVG optimization options */}
+    })
+  ],
+  resolve: {
+    alias: {
```

Comment:
I have mentioned we have alias integration issues that we need to test and come up with a solution. Better leave it from the docs for now.

Reply:
E.g. `@components` should resolve to something like `/your/project/absolute/path/apos-build/@apostrophecms/vite/default/src`

Reply:
Removed. I can work on improving this part later.